rasterio/_err.pyx

```diff
@@ -66,7 +66,7 @@ cdef class GDALErrCtxManager:
         cdef int err_no = CPLGetLastErrorNo()
         cdef const char *msg = CPLGetLastErrorMsg()
         # TODO: warn for err_type 2?
-        if err_type > 0:
+        if err_type > 3:
```

Comment:
Why the change now? Were we catching warnings as exceptions before? And can we use the GDALError enum below?